src_ops_metrics/pre_processing.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019, 2020 Francesco Murdaca
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Pre-processing functions."""
+
+import logging
+
+import numpy as np
+
+from typing import Tuple, Dict, Any, List, Union
+from pathlib import Path
+from datetime import timedelta
+from datetime import datetime
+
+from create_bot_knowledge import load_previous_knowledge
+from utils import convert_num2label, convert_score2num
+from exceptions import MissingPreviousKnowledge
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def retrieve_knowledge(knowledge_path: Path, project: str) -> Union[Dict[str, Any], None]:
+    """Retrieve knowledge (PRs) collected for a project."""
+    project_knowledge_path = knowledge_path.joinpath("./" + f"{project}")
+    pull_requests_data_path = project_knowledge_path.joinpath("./pull_requests.json")
+
+    data = load_previous_knowledge(project, pull_requests_data_path, "PullRequest")
+    if data:
+        return data
+    else:
+        raise MissingPreviousKnowledge(
```

Comment:
In these types of scenarios, shouldn't be more appropriate to log the exception rather than throwing it?